tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -19,13 +19,53 @@
 #include <ROOT/RNTupleMerger.hxx>
 #include <ROOT/RNTupleModel.hxx>
 #include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RPageStorageFile.hxx>
+#include "TFile.h"
 
 Long64_t ROOT::Experimental::RNTuple::Merge(TCollection *inputs, TFileMergeInfo *mergeInfo)
 {
-   if (inputs == nullptr || mergeInfo == nullptr) {
-      return -1;
+   // Check the inputs
+   if (!inputs || inputs->GetEntries() < 3 || !mergeInfo) return -1;
+
+   // Input parameters
+   std::string ntupleName;
+   std::unique_ptr<Detail::RPageSinkFile> destination;
+   std::vector<std::unique_ptr<Detail::RPageSource>> sources;
+   std::vector<Detail::RPageSource *> sourcePtrs;
+
+   // Parse the inputs
+   Long64_t counter {0};
+   TIter itr(inputs);
+   while(const auto& pitr = itr()) {
+      if ( counter == 0 ) {
+         ntupleName = std::string(pitr->GetName());
+      }
+      else if ( counter == 1 ) {
+         TFile* outFile = pitr->InheritsFrom("TFile") ? dynamic_cast<TFile*>(pitr) : nullptr;
+         if(!outFile) return -1;
+         RNTupleWriteOptions writeOpts;
+         writeOpts.SetUseBufferedWrite(false);
+         destination = std::make_unique<Detail::RPageSinkFile>(ntupleName+"_merged", *outFile, writeOpts);
```

Comment:
The `ntupleName+"_merged"` is temporary and creates a new `RNTuple` object in the output file. This needs to be reconciled. Let me add this comment until that's fixed.